remote_consoles.md

```diff
@@ -0,0 +1,12 @@
+### Remote Consoles
+
+Remote consoles allow you to access the screen of a VM through the browser. Each supported protocol has a its own JavaSsript client and they all connect to a common WebSocket endpoint. This endpoint is provided by the Rails application under `/ws/console` and in production it's being served by the `RemoteConsoleWorker`. This worker runs a `RemoteConsole::RackServer` middleware as a web server and it's responsible for proxying and translating between the remote console endpoint and the incoming WebSocket connections.
+
+The server can also be mounted into the development setup of Rails by setting the `MOUNT_REMOTE_CONSOLE_PROXY` environment variable:
+```sh
+MOUNT_REMOTE_CONSOLE_PROXY=1 bin/rails serve
+```
+
+The proxy is implemented using an event loop and the [surro-gate](https://github.com/skateman/surro-gate) gem as the selector for testing the readiness of socket pairs. The translation is done using adapters that are available as subclasses of `RemoteConsole::ServerAdapter` for WebSocket and `RemoteConsole::ClientAdapter` for the client protocols. Implementing a new protocol requires to create a new subclass and update the `.new` method in the related adapter to be able to select it based on the passed context.
+
```

Comment:
^ Maybe we can point them to the directory containing existing protocol implementations?